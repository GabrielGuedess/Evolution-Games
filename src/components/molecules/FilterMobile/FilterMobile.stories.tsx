import * as Dialog from '@radix-ui/react-dialog';
import { Story, Meta } from '@storybook/react/types-6-0';
import { FunnelSimple } from 'phosphor-react';

import { FilterMobile, FilterMobileProps } from './FilterMobile';

export default {
  title: 'Molecules/FilterMobile',
  component: FilterMobile,
  parameters: {
    viewport: {
      defaultViewport: 'mobile1',
    },
  },
  args: {
    filterBy: 'all',
    setFilterBy: () => ({}),
  },
} as Meta<FilterMobileProps>;

export const Default: Story<FilterMobileProps> = args => (
  <Dialog.Root>
    <Dialog.Trigger>
      <FunnelSimple aria-label="Icon" size={32} style={{ cursor: 'pointer' }} />
    </Dialog.Trigger>
    <FilterMobile {...args} />
  </Dialog.Root>
);
