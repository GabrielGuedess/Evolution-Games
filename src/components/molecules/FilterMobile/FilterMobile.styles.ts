import * as Dialog from '@radix-ui/react-dialog';
import media from 'styled-media-query';

import styled, { css } from 'styled-components';

export const Wrapper = styled(Dialog.Content)`
  ${({ theme }) => css`
    position: fixed;
    inset: 0;
    display: flex;
    flex-direction: column;
    overflow-y: auto;
    padding: 3.2rem 2.4rem 7rem 2.4rem;
    background: ${theme.colors.gameDetails};
    z-index: ${theme.layers.alwaysOnTop};

    ${media.greaterThan('large')`
      display: none;
    `}
  `}
`;

export const WrapperContent = styled.div`
  position: relative;
  width: 100%;
  display: flex;
  flex-direction: column;
  gap: 2rem;
`;

export const FooterButton = styled.div`
  position: fixed;
  padding: 0 4rem 2rem 4rem;
  inset: auto 0 0 0;
`;
