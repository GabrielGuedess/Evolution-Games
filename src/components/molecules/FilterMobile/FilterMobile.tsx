import Link from 'next/link';

import * as Dialog from '@radix-ui/react-dialog';

import Button from 'components/atoms/Button/Button';
import { CloseButton } from 'components/atoms/CloseButton/CloseButton';
import {
  Platform,
  PlatformIcon,
} from 'components/atoms/PlatformIcon/PlatformIcon';
import { RadioButton } from 'components/atoms/RadioButton/RadioButton';
import { Title } from 'components/atoms/Title/Title';

import * as S from './FilterMobile.styles';

export type FilterMobileProps = {
  filterBy: Platform;
  setFilterBy: (filterBy: Platform) => void;
};

export const FilterMobile = ({ setFilterBy, filterBy }: FilterMobileProps) => (
  <S.Wrapper>
    <S.WrapperContent>
      <CloseButton side="right" color="white" as={Dialog.Close} />

      <Title size="large" textAlign="left" hasAnimation>
        Platforms
      </Title>

      <form>
        <RadioButton
          title="All"
          id="all"
          value="all"
          name="gameFilter"
          checked={filterBy === 'all'}
          icon={<PlatformIcon platform="all" size="small" />}
          onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
            setFilterBy(e.target.value as Platform)
          }
        />
        <RadioButton
          title="Playstation"
          id="playstation"
          icon={<PlatformIcon platform="playstation" size="small" />}
          value="playstation"
          name="gameFilter"
          checked={filterBy === 'playstation'}
          onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
            setFilterBy(e.target.value as Platform)
          }
        />
        <RadioButton
          title="Xbox"
          id="xbox"
          value="xbox"
          name="gameFilter"
          checked={filterBy === 'xbox'}
          icon={<PlatformIcon platform="xbox" size="small" />}
          onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
            setFilterBy(e.target.value as Platform)
          }
        />
        <RadioButton
          title="PC"
          id="pc"
          value="pc"
          name="gameFilter"
          checked={filterBy === 'pc'}
          icon={<PlatformIcon platform="pc" size="small" />}
          onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
            setFilterBy(e.target.value as Platform)
          }
        />
      </form>

      <S.FooterButton>
        <Link href="/explorer" passHref>
          <Button fullWidth as="a">
            Ver Todos
          </Button>
        </Link>
      </S.FooterButton>
    </S.WrapperContent>
  </S.Wrapper>
);
