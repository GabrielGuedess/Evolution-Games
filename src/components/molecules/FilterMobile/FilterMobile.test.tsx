import * as Dialog from '@radix-ui/react-dialog';
import { renderWithTheme } from 'utils/tests/helpers';

import { fireEvent, screen } from '@testing-library/react';

import { FilterMobile, FilterMobileProps } from './FilterMobile';

let props: FilterMobileProps;

describe('<FilterMobile />', () => {
  beforeEach(() => {
    props = {
      filterBy: 'all',
      setFilterBy: jest.fn(),
    };
  });

  it('should render the component correctly', () => {
    // Arrange
    const { container } = renderWithTheme(
      <Dialog.Root>
        <Dialog.Trigger>Abrir</Dialog.Trigger>
        <FilterMobile {...props} />
      </Dialog.Root>,
    );

    // Act
    fireEvent.click(screen.getByRole('button', { name: 'Abrir' }));

    // Assert
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should change check radio input', () => {
    renderWithTheme(
      <Dialog.Root open>
        <FilterMobile {...props} />
      </Dialog.Root>,
    );

    // Arrange
    const playstation = screen.getByTestId('ps5');
    const xbox = screen.getByTestId('xs');
    const pc = screen.getByTestId('pc');

    // Act
    fireEvent.click(playstation);
    fireEvent.click(pc);
    fireEvent.click(xbox);

    // Assert
    expect(props.setFilterBy).toHaveBeenCalledTimes(3);
  });

  it('should change the default check radio input', () => {
    renderWithTheme(
      <Dialog.Root open>
        <FilterMobile {...props} filterBy="ps5" />
      </Dialog.Root>,
    );

    // Arrange
    const all = screen.getByTestId('all');

    // Act
    fireEvent.click(all);

    // Assert
    expect(props.setFilterBy).toHaveBeenCalledTimes(1);
  });
});
