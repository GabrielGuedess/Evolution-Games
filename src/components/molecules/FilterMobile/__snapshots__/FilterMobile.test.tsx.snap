// Jest Snapshot v1, https://goo.gl/fbAQLP

exports['<FilterMobile /> should render the component correctly 1'] = `
@media (min-width:768px) {

}

@media (min-width:768px) {

}

@media (min-width:1170px) {

}

<button
  aria-controls="radix-:r0:"
  aria-expanded="true"
  aria-haspopup="dialog"
  aria-hidden="true"
  data-aria-hidden="true"
  data-state="open"
  type="button"
>
  Abrir
</button>
`;
